lms/djangoapps/grades/tests/test_models.py

```diff
@@ -0,0 +1,216 @@
+"""
+Unit tests for grades models.
+"""
+import ddt
+from hashlib import md5
+import json
+from mock import patch
+
+from django.db.utils import IntegrityError
+from django.test import TestCase
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.locator import CourseLocator, BlockUsageLocator
+
+from lms.djangoapps.grades.models import BlockRecord, VisibleBlocks, PersistentSubsectionGrade
+
+
+class GradesModelTestCase(TestCase):
+    """
+    Base class for common setup of grades model tests.
+    """
+    def setUp(self):
+        super(GradesModelTestCase, self).setUp()
+        self.course_key = CourseLocator(
+            org='some_org',
+            course='some_course',
+            run='some_run'
+        )
+        self.locator_a = BlockUsageLocator(
+            course_key=self.course_key,
+            block_type='problem',
+            block_id='block_id_a'
+        )
+        self.locator_b = BlockUsageLocator(
+            course_key=self.course_key,
+            block_type='problem',
+            block_id='block_id_b'
+        )
+        self.record_a = BlockRecord(1, 10, self.locator_a)
+        self.record_b = BlockRecord(1, 10, self.locator_b)
+
+
+@ddt.ddt
+class BlockRecordTest(GradesModelTestCase):
+    """
+    Test the BlockRecord model.
+    """
+    def setUp(self):
+        super(BlockRecordTest, self).setUp()
+
+    @ddt.data(True, False)
+    def test_creation(self, use_locator_as_string):
+        """
+        Tests creation of a BlockRecord, both with a locator object and its string representation.
+        """
+        weight = 1
+        max_score = 10
+        record = BlockRecord(
+            weight,
+            max_score,
+            # pylint: disable=protected-access
+            self.locator_a._to_string() if use_locator_as_string else self.locator_a
```

Comment:
Further details: if we change this serialization, I'll also need to update the serialization that takes place in the `locator` property of `BlockRecord`, which currently uses `_from_string()`.
```
======================================================================
FAIL: test_creation_1_True (lms.djangoapps.grades.tests.test_models.BlockRecordTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/ddt.py", line 114, in wrapper
    return func(self, *args, **kwargs)
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/grades/tests/test_models.py", line 63, in test_creation
    self.assertEqual(record.locator, self.locator_a)
AssertionError: BlockUsageLocator(CourseLocator(u'block-v1:some_org', u'some_course', u'some_run', None, None), u'problem', u'block_id_a') != BlockUsageLocator(CourseLocator('some_org', 'some_course', 'some_run', None, None), 'problem', 'block_id_a')
```